docs/widget.rst

```diff
@@ -253,16 +253,25 @@ See also the documentation of the ``resource_registry`` argument in
 Specifying Widget Requirements
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-When creating a new widget, you may specify its requirements by using
-the ``requirements`` attribute:
+When instantiating a new widget, you may specify its requirements by using the ``requirements`` attribute.
+There are two acceptable forms for the specification: two-tuples and dicts.
+The two-tuple form is used by all core Deform widgets and uses the resource registry, taking advantage of its abstraction layer.
+The dict form bypasses the resource registry.
+The dict form may be easier to implement than the two-tuple form for custom widgets.
```

Comment:
```suggestion
When instantiating a new widget, you may specify its requirements by using the ``requirements`` attribute.
The requirements are specified as a sequence of two-tuples and/or dicts.
The two-tuple form is used by all core Deform widgets and uses the resource registry, taking advantage of its abstraction layer.
The dict form bypasses the resource registry.
The dict form may be easier to implement than the two-tuple form for custom widgets, as it does not introduce an implicit dependency on the registry. 
```

To be strictly correct the requirements is a sequence. The items in the sequence must be dicts or two-tuples.

Reply:
Can they be a mixture of both types? I assumed that they are mutually exclusive, but after looking again at the code, it appears that they can be mixed.

Reply:
yes, technically it can be mixed. Theoretically you an require some general version of jQuery from the registry and then some specific code for your widget

Reply:
Got it. One more commit, and one more review from @tdamsma, and I think we can :shipit:!

See 0d4683a.

Reply:
Great!